### PR TITLE
Docs/update husky docs to make them more intuitive

### PR DESCRIPTION
**Pull request by doublethefish**
Initially, I found it quite hard to understand the husky docs, so I thought I would capture my learning and attempt to make the docs easier to follow. Mainly because I think I will be using husky a lot in the future and I have the memory of a goldfish.

I've clarified:
* husky requires git 2.9+ because of `core.hooksPath` support
* what `prepare` does and why
* what the differences between `husky-init` and `husky install` are
* how to manage hooks code
* grouped and added some more comprehensive examples
* added some notes on how to avoid gotchas (in addition to the troubleshooting section)

I've also conformed the 3 readmes, copying the quick-setup sections from the main docs:
* ./README.md
* ./.github/README.md

Finally, I've also added an example `prepare-commit-msg` hook with an external script, as requested by [jcollum-nutrien](https://github.com/jcollum-nutrien)

### CODE REVIEW

**Comment by devinrhode2** [comment]
This looks like really great stuff! I just had a quick glance at files changed

There are some conflicts, could you rebase into latest master "git pull origin master --rebase" to resolve conflicts?
You can consult "git rebase --show-current-patch" to see what a conflicting commit was intending to do

**Comment by wsuwt** [comment]
Thank you! I find this version of documentation a lot less confusing.

**Comment by doublethefish** [comment]
Done.

The conflict was with the instruction to auto-install hooks on `npm install`:
`npm set-script prepare "husky install"` -> `npm pkg set scripts.prepare="husky install"`

I've replaced all three places that describe that step.

**Comment by blwinters** [comment]
@doublethefish It looks like the example for `prepare-commit-msg` is no longer in this PR. Maybe it was accidentally removed during the rebase?

**Comment by doublethefish** [comment]
I removed it. Whilst trying to understand the (very poorly structured) feedback I found a few issues that required some more work, so because it polluted the main purpose of the PR I pulled it for now - I assumed it was slowing reviews. 

I am fairly sure I found a bug in how husky handles bash expansions that would need to be fixed before adding an example. But I'd need to dig into it a produce a decent enough bug report.

Once this gets merged I'll open Issues/PRs for both the bug(s?) and add an example or two, when I get time.

**Comment by blwinters** [comment]
I see. I was able to find a link to your original example that got me unblocked with this, so thanks for that. I'm sure others would be grateful for this to be better explained in the documentation.

**Comment by blwinters** [comment]
You might also consider including checks for "special" commits in your example, like these:

```
COMMIT_SOURCE=$2

# ignore merge commits
if [ "${COMMIT_SOURCE}" = merge ]; then
    exit 0
fi

# do not prefix fixup! squash! commits
if cat "$1" | grep -E -i "^(fixup|squash)!" > /dev/null; then
    exit 0
fi

```

**Comment by doublethefish** [comment]
That's a good suggestion.

File: .github/README.md [hunk 1]
```diff
@@ -8,34 +8,64 @@ Husky improves your commits and more 🐶 _woof!_
...
+`package.json` to use one of the [npm/yarm lifecyle
+hooks](https://docs.npmjs.com/cli/v8/using-npm/scripts#prepare-and-prepublish)
+e.g. `prepare`.
+
 
-```sh
-git commit -m "Keep calm and commit"
-# `npm test` will run every time you commit
+```shell
+npm set-script prepare "husky install"
```
Comment (devinrhode2) [inline .github/README.md:56]:
```suggestion
npm pkg set scripts.prepare="husky install"
```

**Comment by doublethefish** [comment]
@paescuj couldn't agree more but the idea was rejected here #1166 and here #1167.

Personally, I have decided to stop contributing to this project.

**Comment by paescuj** [comment]
Oh, I see. That is really a pity!

**Pull request closed without merging**